joern-cli/frontends/csharpsrc2cpg/src/test/scala/io/joern/csharpsrc2cpg/querying/ast/TypeDeclTests.scala

```diff
@@ -115,4 +118,109 @@ class TypeDeclTests extends CSharpCode2CpgFixture {
       }
     }
   }
+
+  "basic enum types" should {
+
+    val cpg = code(
+      """
+        |enum Season
+        |{
+        |    Spring,
+        |    Summer,
+        |    Autumn,
+        |    Winter
+        |}
+        |""".stripMargin,
+      "Season.cs"
+    )
+
+    "generate a type declaration enum members" in {
+      inside(cpg.typeDecl.nameExact("Season").headOption) {
+        case Some(season) =>
+          season.fullName shouldBe "Season"
+          inside(season.member.l) {
+            case xs if xs.isEmpty => fail("No enum members found!")
+            case spring :: summer :: autumn :: winter :: Nil =>
+              spring.name shouldBe "Spring"
+              spring.code shouldBe "Spring"
+              spring.typeFullName shouldBe DotNetTypeMap(BuiltinTypes.Int)
+
+              summer.name shouldBe "Summer"
+              summer.code shouldBe "Summer"
+              summer.typeFullName shouldBe DotNetTypeMap(BuiltinTypes.Int)
+
+              autumn.name shouldBe "Autumn"
+              autumn.code shouldBe "Autumn"
+              autumn.typeFullName shouldBe DotNetTypeMap(BuiltinTypes.Int)
+
+              winter.name shouldBe "Winter"
+              winter.code shouldBe "Winter"
+              winter.typeFullName shouldBe DotNetTypeMap(BuiltinTypes.Int)
+            case _ => fail("Unexpected number of enum members!")
+          }
+        case None => fail("Unable to find `Season` type decl node")
+      }
+    }
+  }
+
+  "enum types cast as an integer type" should {
+
+    val cpg = code("""
+                    |enum ErrorCode : ushort
+                    |{
+                    |    None = 0,
+                    |    Unknown = 1,
+                    |    ConnectionLost = 100,
+                    |    OutlierReading = 200
+                    |}
+                    |
+                    |""".stripMargin)
+
+    "generate a type declaration enum members" in {
+      inside(cpg.typeDecl.nameExact("ErrorCode").headOption) {
+        case Some(errCode) =>
+          errCode.fullName shouldBe "ErrorCode"
+          inside(errCode.member.l) {
+            case xs if xs.isEmpty => fail("No enum members found!")
+            case none :: unknown :: connectionLost :: outlierReading :: Nil =>
+              none.name shouldBe "None"
+              none.code shouldBe "None = 0"
```

Comment:
So the assignment does not actually end up as assignment call in the CPG?

Reply:
The `<clinit>` method won't be created yet, so there is nothing to test. This test is currently ignored.